### PR TITLE
github:feature added issue/pr templates

### PR DESCRIPTION
**Pull request by nopeslide**
I would like to have various templates to speed up issue/pr filing.
open to suggestions, feel free to commit into this branch.

### CODE REVIEW

**Review by ikstream — COMMENTED** [review]
I am not sure what's the difference between clean workspace/build/container

`make clean; make all` would fit all, wouldn't it?

Assignees is something you would need to set and edit the issue. An assignment can be done through GH internal and don't need 2 be in the issue head, don't you think?

Otherwise LGTM

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+**Screenshots/Code**
+If applicable, add screenshots or code to help explain your problem.
+
+**Additional context**
+Add any other context about the problem here.
+
+**Checklist**
+- [ ] clean build
+- [ ] clean workspace
+- [ ] clean container
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/bug.md]:
```suggestion
- [ ] git pull
- [ ] make clean
- [ ] make dirclean
- [ ] make distclean
```

Comment (nopeslide) [reply .github/ISSUE_TEMPLATE/bug.md]: > Assignees is something you would need to set and edit the issue. An assignment can be done through GH internal and don't need 2 be in the issue head, don't you think?

didn't touch this, it was autogenerated by github

**Review by ikstream — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/bugfix.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+**Related**
+Name the the issues, this bugfix solves.
+
+**Checklist**
+- [ ] issue was filed
+- [ ] clean build
+- [ ] clean workspace
+- [ ] clean container
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/bugfix.md]:
```suggestion
- [ ] tests added
- [ ] documentation added
```

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Name related PR and issues.
+
+**Description**
+A clear and concise description of what you want to happen.
+
+**Checklist**
+- [ ] issue was filed
+- [ ] clean build
+- [ ] clean workspace
+- [ ] clean container
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/feature.md]:
```suggestion
- [ ] tests added
- [ ] documentation added 
```

**Review by ikstream — CHANGES_REQUESTED** [review]
I would remove `tested` and `compiles without warnings` this is something we should expect from every pull request. If it doesn't it should be explicitly stated in the pull request with reasoning why. We should build everything by our self in addition to the CI test before merging it into upstream to be safe anyway.

**Comment by nopeslide** [comment]
> I would remove `tested` and `compiles without warnings` this is something we should expect from every pull request. If it doesn't it should be explicitly stated in the pull request with reasoning why. We should build everything by our self in addition to the CI test before merging it into upstream to be safe anyway.

@ikstream 
What about PR Drafts and on device testing?
A PR should conform to all of these items, they act as reminders.
If your code does not compile: we have an unchecked box that forces an explanation.

**Comment by ikstream** [comment]
> What about PR Drafts and on device testing?
> A PR should conform to all of these items, they act as reminders.
> If your code does not compile: we have an unchecked box that forces an explanation.

Don't we have compile tests? If they fail there is some explanation anyways?

File: .github/ISSUE_TEMPLATE/feature.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+name: feature
+about: Suggest an idea for this project
+title: 'feature: '
+labels: enhancement
+assignees: ''
+
+---
+
+**Is your feature request related to a problem? Please describe.**
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/feature.md]:
```suggestion
**Description**
```

File: .github/ISSUE_TEMPLATE/feature.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+
+**Describe the solution you'd like**
+A clear and concise description of what you want to happen.
+
+**Describe alternatives you've considered**
+A clear and concise description of any alternative solutions or features you've considered.
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/feature.md]:
```suggestion
**Alternative solutions**
A clear and concise description of any alternative solutions or features you've considered.
- first alternative solution
- second alternative solution
```

File: .github/ISSUE_TEMPLATE/feature.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+labels: enhancement
+assignees: ''
+
+---
+
+**Is your feature request related to a problem? Please describe.**
+A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+
+**Describe the solution you'd like**
+A clear and concise description of what you want to happen.
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/feature.md]:
```suggestion
**Solution**
A clear and concise description of your preferred solution.
```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. See error
+
+**Screenshots/Code**
+If applicable, add screenshots or code to help explain your problem.
+
+**Additional context**
+Add any other context about the problem here.
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/bug.md]:
```suggestion
**Context**
If applicable, add screenshots or code to help explain your problem and any other context about the problem here.
```

File: .github/ISSUE_TEMPLATE/feature.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+**Describe alternatives you've considered**
+A clear and concise description of any alternative solutions or features you've considered.
+
+**Additional context**
+Add any other context or screenshots about the feature request here.
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/feature.md]:
```suggestion
**Context**
If applicable, add screenshots or code to help explain your problem and any other context about the problem here.
```

File: .github/PULL_REQUEST_TEMPLATE/bugfix.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+name: bugfix
+about: Request to merge a bugfix
+title: 'bugfix: '
+labels: bug
+assignees: ''
+
+---
+
+**Related**
+Name the the issues, this bugfix solves.
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/bugfix.md]:
```suggestion
**Related Issues & PRs**
Name the issues & PRs, this bugfix solves or relates to.
If no issue exists, please create one!
- #??
- #??
```

File: .github/PULL_REQUEST_TEMPLATE/bugfix.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+**Checklist**
+- [ ] issue was filed
+- [ ] tests added
+- [ ] documentation added
+- [ ] no related PR
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/bugfix.md]:
```suggestion
- [ ] tests added
- [ ] documentation added
```

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+name: feature
+about: Request to merge a feature
+title: 'feature: '
+labels: enhancement
+assignees: ''
+
+---
+
+**Related**
+Name related PR and issues.
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/feature.md]:
```suggestion
**Related Issues & PRs**
Name related PR and issues.
If no issue exists, please create one!
- #??
- #??
```

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+**Description**
+A clear and concise description of what you want to happen.
+
+**Checklist**
+- [ ] issue was filed
+- [ ] tests added
+- [ ] documentation added 
+- [ ] no related PR
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/feature.md]:
```suggestion
- [ ] tests added
- [ ] documentation added 
```

File: .github/PULL_REQUEST_TEMPLATE/feature.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+assignees: ''
+
+---
+
+**Related**
+Name related PR and issues.
+
+**Description**
+A clear and concise description of what you want to happen.
+
```
Comment (nopeslide) [inline .github/PULL_REQUEST_TEMPLATE/feature.md:18]:
```suggestion

**Context**
If applicable, add screenshots or code to help explain your problem and any other context about the problem here.

```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+**Checklist**
+- [ ] git pull
+- [ ] make clean
+- [ ] make dirclean
+- [ ] make distclean
+- [ ] no related issue
```
Comment (nopeslide) [inline .github/ISSUE_TEMPLATE/bug.md]:
```suggestion
```

**Review by nopeslide — COMMENTED** [review]
Simplified everything a little bit

File: .github/ISSUE_TEMPLATE/bug.md (hunk 14 above)
Comment (nopeslide) [reply .github/ISSUE_TEMPLATE/bug.md]: either the bug is reproducable or not.
A check on related issues has no meaning at all

**Review by ikstream — DISMISSED** [review]

**Review by ikstream — APPROVED** [review]
lgtm

**Pull request merged**